src/pages/landing/index.test.tsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import { http, HttpResponse, delay } from 'msw';
+import { setupServer } from 'msw/node';
+import { screen, fireEvent } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom';
+
+import { renderWithProviders } from 'utils/test-utils';
+import LandingPage from './';
+
+const server = setupServer(
+  http.get('/server-timestamp', async () => {
+    await delay(150);
+    return HttpResponse.json(1);
+  }),
+  http.post('/login', async () => {
+    await delay(150);
+    return HttpResponse.json('auth-token');
+  }),
+  http.get('/profile/current/subscriptions/active-features', async () => {
+    await delay(150);
+    return HttpResponse.json([]);
+  }),
+  http.get('/tiers', async () => {
+    await delay(150);
+    return HttpResponse.json([{
+      id: '1',
+      name: 'Tier 1',
+      subtitle: 'Tier subtitle',
+      features: [
+        { name: 'Feature 1', },
+        { name: 'Feature 2', }
+      ],
+      usdPrice: 100,
+      enabled: true,
+    }]);
+  }),
+  http.get('/ada-usd-price', async () => {
+    await delay(150);
+    return HttpResponse.json(10.0);
+  }),
+  http.put('/profile/current', async () => {
+    await delay(150);
+    return HttpResponse.json({});
+  }),
+  http.get('/profile/current/subscriptions', async () => {
+    await delay(150);
+    return HttpResponse.json([{ id: '1', status: 'pending', price: 10 }]);
+  }),
+  http.post('/profile/current/subscriptions/1', async () => {
+    await delay(150);
+    return HttpResponse.json({ id: '1' }, { status: 201 });
+  }),
+  http.get('/profile/current/balance', async () => {
+    await delay(150);
+    return HttpResponse.json(-1);
+  }),
+);
+
+beforeAll(() => server.listen());
+afterEach(() => server.resetHandlers());
+afterAll(() => server.close());
+
+jest.mock('@emurgo/cardano-serialization-lib-browser', () => ({
+  Address: {
+    from_bytes: () => 'address',
+  },
+  BaseAddress: {
+    from_address: () => ({
+      stake_cred: () => 'credential',
+      to_address: () => ({
+        to_bech32: () => 'address',
+      }),
+    }),
+  },
+  RewardAddress: {
+    new: () => ({
+      to_address: () => ({
+        to_hex: () => 'address',
+        to_bech32: () => 'address',
+      }),
+    }),
+  },
+  BigNum: {
+    from_str: (a: string) => ({
+      less_than: (b: any) => parseFloat(a) < b.get(),
+      checked_sub: (b: any) => ({
+        less_than: (c: any) => (parseFloat(a) - b.get()) < c.get(),
+      }),
+      get: () => parseFloat(a),
+    })
+  }
+}));
+
+jest.mock('store/slices/walletTransaction.slice', () => ({
+  payFromWallet: () => (() => ({ payload: 'transaction-id' }))
+}));
+
```

Comment:
Also keep mocks into a separate file , for easiness of maintainability

Reply:
I agree, but I'm struggling to make it work. Can we separate it in another Jira task?